### PR TITLE
Sampler refactor

### PR DESCRIPTION
**Pull request by alanakbik**
This PR adds a `FlairSampler` interface to better enable passing custom samplers to the `ModelTrainer`. 

For instance, if you want to always shuffle your dataset in chunks of 5 to 10 sentences, you provide a sampler like this: 

```python
# your trainer
trainer: ModelTrainer = ModelTrainer(tagger, corpus)

# execute training run
trainer.train('path/to/experiment/folder',
              max_epochs=150,
              # sample data in chunks of 5 to 10
              sampler=ChunkSampler(block_size=5, plus_window=5)
              )
```

### CODE REVIEW

**Comment by yosipk** [comment]
👍

**Comment by alanakbik** [comment]
:+1:
